openssl/src/hash.rs

```
@@ -1,3 +1,51 @@
+//! An abstract interface over various specific cryptographic hash algorithms.
+//!
+//! The `Hasher` type can be configured to hash bytes with any of a variety of supported hash
+//! algorithms. It is particularly useful in cases where you need to work with any type
+//! that supports a hashing interface. Furthermore, `Hasher` implements the [std::io::Write](
+//! https://doc.rust-lang.org/std/io/trait.Write.html) trait.
+//! # Examples
+//!
+//! ```rust
+//! extern crate openssl;
+//! extern crate hex;
+//! 
+//! use openssl::hash;
+//! use hex::ToHex;
+//! 
+//! fn main() {
+//!     let mut hasher = hash::Hasher::new(hash::MessageDigest::sha512()).unwrap();
+//!     hasher.update(b"HTTP/1.1 200 OK\r\n");
```

We shouldn't be ignoring the errors here.